frontend/packages/dev-console/src/components/pipelines/PipelineDetailsPage.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import { DetailsPage, DetailsPageProps } from '@console/internal/components/factory';
+import { Kebab, navFactory } from '@console/internal/components/utils';
+import { k8sGet, k8sList } from '@console/internal/module/k8s';
+import PipelinEnvironmentComponent from './PipelineEnvironment';
+import PipelineDetails from './PipelineDetails';
+import PipelineRuns from './PipelineRuns';
+import { triggerPipeline, rerunPipeline } from '../../utils/pipeline-actions';
+import { getLatestRun } from '../../utils/pipeline-augment';
+import { PipelineRunModel, PipelineModel } from '../../models';
+
+interface PipelineDetailsPageStates {
+  menuActions: Function[];
+}
+
+class PipelineDetailsPage extends React.Component<DetailsPageProps, PipelineDetailsPageStates> {
+  constructor(props) {
+    super(props);
+    this.state = { menuActions: [] };
+  }
+
+  componentDidMount() {
+    // eslint-disable-next-line promise/catch-or-return
+    k8sGet(PipelineModel, this.props.name, this.props.namespace).then((res) => {
+      // eslint-disable-next-line promise/no-nesting, promise/catch-or-return
+      k8sList(PipelineRunModel, {
+        labelSelector: { 'tekton.dev/pipeline': res.metadata.name },
+      }).then((listres) => {
+        this.setState({
+          menuActions: [
+            triggerPipeline(res, getLatestRun({ data: listres }, 'creationTimestamp'), 'pipelines'),
```

Comment:
We should add RBAC checks to the menu items. For example,

https://github.com/openshift/console/blob/master/frontend/public/components/utils/kebab.tsx#L65-L78